drivers/led/sn32/rgb_matrix_sn32f24xb.c

```diff
@@ -0,0 +1,424 @@
+
+            case C12:
+                cfg->channels[20].pfpamsk = 1;
+            case B4:
+                cfg->channels[20].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i]      = 20;
+                break;
+
+            case C13:
+                cfg->channels[21].pfpamsk = 1;
+            case B5:
+                cfg->channels[21].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i]      = 21;
+                break;
+
+            case C14:
+                cfg->channels[22].pfpamsk = 1;
+            case B6:
+                cfg->channels[22].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i]      = 22;
+                break;
+
+            case C15:
+                cfg->channels[23].pfpamsk = 1;
+            case B7:
+                cfg->channels[23].mode = PWM_OUTPUT_ACTIVE_LOW;
+                chan_col_order[i]      = 23;
+                break;
+        }
+    }
+}
+static void rgb_callback(PWMDriver *pwmp);
+
+static void shared_matrix_rgb_enable(void) {
+    pwmcfg.callback = rgb_callback;
+    pwmEnablePeriodicNotification(&PWMD1);
+}
+
+static void shared_matrix_rgb_disable_pwm(void) {
+    // Disable PWM outputs on column pins
+    for (uint8_t y = 0; y < LED_MATRIX_COLS; y++) {
+        pwmDisableChannelI(&PWMD1, chan_col_order[y]);
+    }
+}
+
+static void shared_matrix_rgb_disable_leds(void) {
+    // Disable LED outputs on RGB channel pins
+    for (uint8_t x = 0; x < LED_MATRIX_ROWS_HW; x++) {
+        writePinLow(led_row_pins[x]);
+    }
+}
+
+static void update_pwm_channels(PWMDriver *pwmp) {
+    matrix_row_t row_shifter = MATRIX_ROW_SHIFTER;
+    for (uint8_t col_idx = 0; col_idx < LED_MATRIX_COLS; col_idx++, row_shifter <<= 1) {
+#ifdef MATRIX_NO_SCAN
+#    if (DIODE_DIRECTION == ROW2COL)
+        // Scan the key matrix column
+        if (!matrix_scanned) {
+            matrix_locked = true;
+            matrix_read_rows_on_col(shared_matrix, col_idx, row_shifter);
+        }
+#    endif // DIODE_DIRECTION == ROW2COL
+#endif     // MATRIX_NO_SCAN
+        uint8_t led_index = g_led_config.matrix_co[row_idx][col_idx];
+        // Check if we need to enable RGB output
+        if (led_state[led_index].b != 0) enable_pwm |= true;
+        if (led_state[led_index].g != 0) enable_pwm |= true;
+        if (led_state[led_index].r != 0) enable_pwm |= true;
+        // Update matching RGB channel PWM configuration
+        switch (current_row % LED_MATRIX_ROW_CHANNELS) {
+            case 0:
+                if (enable_pwm) pwmEnableChannelI(pwmp, chan_col_order[col_idx], led_state[led_index].b);
+                break;
+            case 1:
+                if (enable_pwm) pwmEnableChannelI(pwmp, chan_col_order[col_idx], led_state[led_index].g);
+                break;
+            case 2:
+                if (enable_pwm) pwmEnableChannelI(pwmp, chan_col_order[col_idx], led_state[led_index].r);
+                break;
+            default:;
+        }
+    }
```

Comment:
If the above loop is present here in its current state, it would scan the whole `ROW2COL` matrix. So with the other suggested change for managing `matrix_locked` and `matrix_scanned` for `COL2ROW` something like this needs to be added here:
```suggestion
    }
#ifdef MATRIX_NO_SCAN
#    if (DIODE_DIRECTION == ROW2COL)
        // The whole matrix has been scanned
        matrix_locked = false;
        matrix_scanned = true;
#    endif // DIODE_DIRECTION == ROW2COL
#endif     // MATRIX_NO_SCAN
```
(No point checking for any conditions here, because if `matrix_scanned` was not `false` at the start of this function, that code would just set the variables to the same values as they were.)